src/cryptography/hazmat/backends/openssl/x509.py

```diff
@@ -534,3 +534,22 @@ def entry_type(self):
         # we only have precerts.
         assert entry_type == self._backend._lib.CT_LOG_ENTRY_TYPE_PRECERT
         return x509.certificate_transparency.LogEntryType.PRE_CERTIFICATE
+
+    def __hash__(self):
+        return hash(
+            (self.version, self.log_id, self.timestamp, self.entry_type)
```

Comment:
This isn't right, we don't expose enough info ATM to reliably compare these.

Reply:
If we bound `SCT_get0_signature` we could use signature bytes as a comparator, right?

Reply:
I guess signature would probably be enough